includes/functions/updates/llms-functions-updates-4140.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Update functions for version 4.14.0
+ *
+ * @package LifterLMS/Functions/Updates
+ *
+ * @since [version]
+ * @version [version]
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Remove orphan access plans
+ *
+ * @since [version]
+ *
+ * @return bool True if it needs to run again, false otherwise.
+ */
+function llms_update_4140_remove_orphan_access_plans() {
+
+	$limit = 50;
+	$skip  = get_transient( 'llms_4140_skipper_orphan_access_plans' );
+	if ( ! $skip ) {
+		$skip = 0;
+	}
+	set_transient( 'llms_4140_skipper_orphan_access_plans', $skip + $limit, DAY_IN_SECONDS );
```

Comment:
I haven't modified anything to test this, but I'm pretty sure we don't need (or want) a skipper in this routine.

Since we'll be deleting the results we want to start at 0 again on the next run. If we use a skipper we'll end up deleting 50 and then skipping the next 50 (because the index of the first item in the next 50 items will be 0 during the next query, not 49).

Right?

I think if you write a test with pagination you'll see what I mean. I think. Maybe

Reply:
This happens when you're hungry and tired, and you copy and paste, and mostly you are a very good developer!